### ISSUE TITLE
302 error for every checkout

### ISSUE THREAD

**Issue by crezzur** [mollie/PrestaShop#359]
### Actual behavior
Every-time when you check out a product for paying a call is made to http://storename/module/mollie/payment?method=ideal&rand=1623492603. This always returns an 302 error which i followed by accessing the correct URL https://www.mollie.com/paymentscreen/issuer/select/ideal/########

The 302 error can hold back the payment progress up to 6 seconds.

### Steps to reproduce the behavior
1. Place product in cart
2. Go to checkout
3. Select payment method
4. Press on checkout button (With network monitor open)
5. Call 1: is to payment controller which result in 302 error.
6. Call 2: is made to the correct url https://www.mollie.com/paymentscreen/issuer/select/

### Network monitor screen
![error](https://user-images.githubusercontent.com/17967576/121772982-49870980-cb79-11eb-952d-15ba3048243f.jpg)

### Environment
* PrestaShop version: 1.6.1.24 & 1.7.7.4 
* Mollie module version: 3.3.4 & 4.3.0
* PHP version: 7.1.26 -& 7.3.12

**Comment by Saimis777** [comment]
Hey @crezzur ,

Thank you for your message!

We will check with this issue internally, try to replicate it, and will let you know what we'll find!

Have a nice day!

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hi @crezzur,

This does not even seem like an issue, that URL calls our module which saves required information about the order/payment, creates order in PS and then redirects to Mollie checkout URL.

It can take some time, but that is required to do before redirecting to Mollie checkout URL.

And 302 is not an error code, it’s “Found” and it means pretty much the same as 200 HTTP code. Only 4** and 5** are HTTP error codes.

In short: this does not seem like an issue.

Let me know if you have any further questions or if we can close the ticket!

--
Best regards,
Invertus Support team

**Comment by crezzur** [comment]
After some troubleshooting i have found out that this might cause the issue for my client.
(https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Upgrade)

My client uses HTTP/2 and it seems that Mollie does not support this. The Mollie module fails to connect to the API Server (Prestashop module Mollie 4.3.0)  On the front office this causes a delay of 6 seconds.

The error of the Mollie module 4.3.0 sends a message that the API id is incorrect. There was no issue with Mollie 3.5.0 (except of 6 seconds delay) Mollie 4.3.0 does not work with HTTP/2 (no connection possible)

**Comment by SimonasB88** [comment]
@crezzur hello, and thank you for your submission!
Well, I am passing the issue to the team, for further investigation about this HTTP versioning case.

I will let you know as fast as possible about the investigation status.

Best Regards,
Invertus Support Team

**Comment by SimonasB88** [comment]
Hello @crezzur

From your message we understand that payments do not work for you on 4.3.0 due to message “API id is incorrect” which was not the initial reason for this issue. And there’s no real information from your side regarding this issue for us to reproduce this bug.

Mollie 3.5.0 is a very old version and should not be used as a reference point.

We have as well tested the URL for payment by disabling redirect to check out URL, and it responded with 200 OK HTTP code, which concludes that 302 FOUND is being returned by Prestashop redirect functionality (we have no functionality in redirection). 

Best Regards,
Invertus Suport Team

**Comment by Saimis777** [comment]
Hello @crezzur ,

Do you have any other questions maybe that we could assist you with?

Let us know!

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hello,

Due to inactivity we will close this issue for now.

Let us know if you'll need further assistance!

--
Best regards,
Invertus Support team
